services/components-api-service-rest/src/main/java/org/talend/components/service/rest/impl/PropertiesControllerImpl.java

```
@@ -92,11 +100,41 @@ public String getProperties(String definitionName, String formName) {
     }
 
     @Override
+    public ResponseEntity<InputStreamResource> getIcon(String definitionName, DefinitionImageType imageType) {
+        notNull(definitionName, "Definition name cannot be null.");
+        notNull(imageType, "Definition image type cannot be null.");
+        final Definition<?> definition = propertiesHelpers.getDefinition(definitionName);
+        notNull(definition, "Could not find definition of name %s", definitionName);
+
+        // Undefined and missing icon resources are simply 404.
+        String imagePath = definition.getImagePath(imageType);
+        if (imagePath == null) {
+            return new ResponseEntity<>(HttpStatus.NOT_FOUND);
```

Quick try with single return turns out to be pretty ugly with nested conditionals -- I'm OK with using extra returns to fail fast at the start of the method.  (There are no resources to clean up in any of these methods).